crd-generator/src/main/java/io/fabric8/crd/generator/AbstractCustomResourceHandler.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crd.generator;
+
+import static io.fabric8.crd.generator.utils.Types.findStatusProperty;
+
+import io.fabric8.crd.generator.apt.CustomResourceInfo;
+import io.fabric8.crd.generator.apt.Resources;
+import io.fabric8.crd.generator.decorator.Decorator;
+import io.fabric8.crd.generator.utils.Strings;
+import io.fabric8.crd.generator.visitor.AdditionalPrinterColumnDetector;
+import io.fabric8.crd.generator.visitor.LabelSelectorPathDetector;
+import io.fabric8.crd.generator.visitor.SpecReplicasPathDetector;
+import io.fabric8.crd.generator.visitor.StatusReplicasPathDetector;
+import io.sundr.codegen.functions.ElementTo;
+import io.sundr.codegen.model.Property;
+import io.sundr.codegen.model.TypeDef;
+import io.sundr.codegen.model.TypeDefBuilder;
+import io.sundr.codegen.model.TypeRef;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.lang.model.element.TypeElement;
+
+public abstract class AbstractCustomResourceHandler {
```

Comment:
I would really appreciate it if you could add JavaDoc for each of these classes explaining how they fit together for the benefit of future developers.

Reply:
Done